.circleci/config.yml

```
@@ -471,40 +487,30 @@ jobs:
       CC: clang
       CXX: clang++
       MAKEFLAGS: -j 10
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
+    <<: *steps_b_ubu
 
-  b_ubu_asan_clang: &b_ubu_asan_clang
-    docker:
-      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
-    environment:
-      CC: clang
-      CXX: clang++
-      CMAKE_OPTIONS: -DSANITIZE=address
-      MAKEFLAGS: -j 3
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
-
-  b_ubu_ubsan_clang: &b_ubu_ubsan_clang
+  b_ubu_au_clang:
+    parameters:
+      cmake_option:
+        type: string
+      gitter_notify:
+        type: boolean
+        default: false
     docker:
       - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
     environment:
       CC: clang
       CXX: clang++
```

We could also make this a parameter. This way we'd have one job for both GCC and clang:

```suggestion
      CC: <<# parameters.clang >> clang <</ parameters.clang >><<^ parameters.clang >> gcc <</ parameters.clang >>
      CXX: <<# parameters.clang >> clang++ <</ parameters.clang >><<^ parameters.clang >> g++ <</ parameters.clang >>
```

I wonder if CircleCI has more concise syntax for [boolean parameters](https://circleci.com/docs/2.0/reusing-config/#boolean) though.

---

This would also require an option for `MAKEFLAGS: -j 3`. It's 10 for the clang job because it runs on `resource_class: xlarge` so it can benefit from more cores.

In the worst case though, I think that having 10 in both cases would not be that bad. It would not get any speedup over 3 but it should not make it noticeably slower.

---

Oh, I see that it uses a different base image. So it would have to be another parameter.

Or you could leave it as is but instead unify `b_ubu_au_clang` and `b_ubu_clang`. Adding an empty `CMAKE_OPTIONS` in `b_ubu_clang` should not hurt.